src/Product/Repositories/ProductRepository.php

```diff
@@ -2,47 +2,89 @@
 
 namespace IndieHD\Velkart\Product\Repositories;
 
-use Illuminate\Database\Eloquent\Collection;
+use IndieHD\Velkart\Base\Repositories\BaseRepository;
+use IndieHD\Velkart\Base\Traits\UploadsFiles;
 use IndieHD\Velkart\Product\Contracts\ProductRepositoryContract;
 use IndieHD\Velkart\Product\Product;
+use IndieHD\Velkart\ProductImage\Contracts\ProductImageRepositoryContract;
+use Illuminate\Database\DatabaseManager;
 
-class ProductRepository Implements ProductRepositoryContract
+class ProductRepository extends BaseRepository Implements ProductRepositoryContract
 {
+    use UploadsFiles;
+
     /**
      * @var Product
      */
-    protected $model;
+    protected $product;
+
+    /**
+     * @var ProductImageRepositoryContract
+     */
+    protected $productImage;
+
+    /**
+     * @var DatabaseManager
+     */
+    protected $db;
 
     /**
      * ProductRepository constructor.
      * @param Product $product
+     * @param ProductImageRepositoryContract $productImage
+     * @param DatabaseManager $db
      */
-    public function __construct(Product $product)
+    public function __construct(Product $product, ProductImageRepositoryContract $productImage, DatabaseManager $db)
     {
-        $this->model = $product;
+        $this->product = $product;
+        $this->productImage = $productImage;
+        $this->db = $db;
     }
 
-    /**
-     * List all the products
-     *
-     * @param string $order
-     * @param string $sort
-     * @param array $columns
-     * @return Collection
-     */
-    public function list(string $order = 'id', string $sort = 'desc', array $columns = ['*']): Collection
+    public function model(): object
     {
-        return $this->model->orderBy($order, $sort)->get($columns);
+        return $this->product;
+    }
+
+    public function modelClass(): string
+    {
+        return Product::class;
+    }
+
+    public function delete(int $id): bool
+    {
+        $model = $this->model()->find($id);
+        $model->images()->delete();
+        return $model->delete();
```

Comment:
It seems like this should be wrapped in a transaction. As it is now, it's possible that all the images are deleted, but then the product fails to be deleted, thereby leaving the data in an inconsistent state.